### PR TITLE
[quant] Add default symmetric qat qconfig for qnnpack

### PR DESCRIPTION
**Pull request by digantdesai**
Summary:
* This is the default symmetric qat qconfigs for qnnpack.
* Support for symmetric quantization is not available from other backends.
* Observers are similar to symmetric PTQ qconfigs for qnnpack.

Test Plan:
* Added new tests `test_conv_linear_symm` under `quantization.eager.test_quantize_eager_qat.TestQuantizeEagerQAT`, and `test_linear_bn_symm_numerics` under `quantization.eager.test_quantize_eager_qat.TestQuantizeEagerQATNumerics`
    * Only for qnnpack qengine
    * Skipping this test if XNNPACK is not available

Differential Revision: D34804808

### CODE REVIEW

File: test/quantization/eager/test_quantize_eager_qat.py [hunk 1]
```diff
@@ -340,6 +343,37 @@ def checkQuantized(model):
                 model = quantize_qat(model, test_only_train_fn, [self.img_data_2d_train])
                 checkQuantized(model)
 
+    @skipIfNoXNNPACK
+    def test_conv_linear_symm(self):
```
Comment (jerryzh168) [inline test/quantization/eager/test_quantize_eager_qat.py:347]: do we need to add a test for numerics as well?

File: test/quantization/eager/test_quantize_eager_qat.py [hunk 2]
```diff
@@ -51,6 +52,8 @@
     override_qengines,
 )
 
+from torch.testing._internal.common_utils import skipIfNoXNNPACK
```
Comment (jerryzh168) [inline test/quantization/eager/test_quantize_eager_qat.py:55]: I thought we are going to hide xnnpack entirely, are we still exposing this in torch.backends?

**Review by jerryzh168 — APPROVED** [review]
looks good, had some comments inline

**Comment by jerryzh168** [comment]
can we update the summary and test plan here as well?

File: test/quantization/eager/test_quantize_eager_qat.py (hunk 2 above)
Comment (digantdesai) [reply test/quantization/eager/test_quantize_eager_qat.py:55]: yes, that's the goal. I wish I didn't have to add this but for OSS CI there are some builds which don't have USE_XNNPACK, in those cases this test will fail. 

Until XNNPACK is optional we have to do this.

File: test/quantization/eager/test_quantize_eager_qat.py (hunk 1 above)
Comment (digantdesai) [reply test/quantization/eager/test_quantize_eager_qat.py:348]: Added! Thanks!

**Pull request closed without merging**